src/store/completion.rs

```diff
@@ -0,0 +1,138 @@
+use std::{ops::Range, sync::Arc, time::Duration};
+
+use crossbeam_channel::{after, Receiver, Sender};
+use io_uring::IoUring;
+
+use crate::node_pages_map::Page;
+
+pub(crate) fn start_completion_handler(
+    ring: Arc<IoUring>,
+    _ring_capacity: u32,
+    rx_wait_completion: Receiver<(Range<u64>, Option<Vec<Box<Page>>>, Sender<()>)>,
+    rx_kill_completion_handler: Receiver<()>,
+) {
+    std::thread::spawn(move || {
+        let mut complete = RangeVector::new();
+        // batched of operations that waits for completition
+        let mut pendings = Vec::<(Range<u64>, Option<Vec<Box<Page>>>, Sender<()>)>::new();
+
+        // TODO: Waking up the completion thread every X microseconds may not be the best solution.
```

Comment:
this seems wrong. the kernel should unpark whenever there's a completion, no? maybe we have to give one thread the submission queue and another the completion queue?

Reply:
Oh and to add to that, you may still better to wait for the completion of the requests, because you are better not to free buffers before that.